Types/StatusCode.php

```diff
@@ -33,11 +33,11 @@ class StatusCode {
      * @var array
      */
     private static $expectedStatusCodes = [
-        'get'    => 200,
-        'put'    => 200,
-        'patch'  => 200,
-        'post'   => 201,
-        'delete' => 204
+        'get'    => [200, 203, 206, 404],
```

Comment:
the logic behind that is for when a request is made to an endpoint such as /entity/1 and a 404 is returned to say it doesn't exist. A database would happily return empty data but this driver would throw an exception. A 404 should be accepted as empty data to allow doctrine to handle it correctly